### PR TITLE
Fixes for 3.38

### LINKED ISSUE #212: Not working for gnome-shell 3.38
I know the project isn't maintained anymore, but just in case

### Linux Distribution version
Debian GNU/Linux bullseye

### Gnome Shell version
3.38

### Xorg or Wayland (or both)
Xorg

### Extension version or branch
53

### Description of the problem
Clutter.Actor API changed rendering the extension unusable.
Note that the code requests Tweener, not used, but also reports an error.

### Steps to reproduce the problem
Just install it and look at the logged errors.

### PR DESCRIPTION
**Pull request by RensAlthuis**
closes #212

Might be some bugs in there still, but at least it appears to be working.

### CODE REVIEW

**Comment by sheemon-lists** [comment]
Old as I am, I am brand spanking new to this Git business, so bear with me, if you can.
This beautiful module blows up (like many, many others) on Gnome 3.38 for the same reason;
The function "tweener" was replaced with another.  I believe an issue #215 referred to it.
This change broke every extension that even lists it.
If you already know about it and corrected the bug, then thank you so much and please tell me how I can get an early copy.
If you think I should fix this "issue", then please tell me how.
Please do not be concerned.  I have seen programming before :-)
What I am totally at loss, is all the procedures around the code.
Not to mention I have no idea how the Gnome shell is written nor built.
But perhaps I can learn, given a chance.

Awaiting your reply.
Simon

**Comment by horga83** [comment]
This has already all been fixed in this fork:

https://github.com/RensAlthuis/workspaces-to-dock

**Comment by dpriedel** [comment]
Hi

Thanks for updating this extension.  It is my all-time most useful gnome extension.

All seems to work well except for 1 thing.  In overview mode, I can't drag a thumbnail icon from 1 workspace to another to move the application window.  I can drag the window itself though.

Some more information.

There are numerous entries in the system journal file such as

Oct 15 09:40:16 Leaphorn gnome-shell[12876]: Can't update stage views actor Gjs_workspaces-to-dock_passingthru67_gmail_com_myWorkspaceThumbnail_WorkspacesToDock_MyWorkspaceThumbnail is on because it needs an allocation.
Oct 15 09:40:16 Leaphorn gnome-shell[12876]: Can't update stage views actor workspacestodockCaptionBackground is on because it needs an allocation.

I don't know what this means but it seems there is some non-fatal problem with the code...

Thanks !!!

**Comment by RensAlthuis** [comment]
I fixed the thumbnail icons not working properly, that was an easy fix. 

I'm not sure what the allocation errors are about. These were already present before the 3.38 update. I also can't find any documentation on how allocation is supposed to work (or what it does.. I think it's the extension claiming a part of the screen for it's own use but that's just a guess at this point). It seems to be working regardless of those errors so I'm going to ignore them for now.

**Comment by dpriedel** [comment]
Except for the extra journal data this extensions seems to be working fine.

Thanks !!!

**Comment by xakraz** [comment]
Thx @RensAlthuis !
:pray:

**Comment by Blackrucat** [comment]
Put it on https://extensions.gnome.org/
pls, I am very newbie, and I don't know how to install this one.

**Comment by Blackrucat** [comment]
https://github.com/RensAlthuis/workspaces-to-dock/issues/5#issuecomment-710608353
This helped me a lot, thanks.
but anyways, I'm still thinking it's nice to put this on 
https://extensions.gnome.org/

**Comment by RensAlthuis** [comment]
@Blackrucat You're welcome, It's good to hear this is helping others! 
Regarding my fork going on the website: For now I'm waiting on Gnome 40, it will completely break this extension. On top of that, I may have to drop gnome as my window manager since Gnome 40 will remove the choice between vertical and horizontal workspaces, it will force everyone to use the horizontal layout instead and well... vertical workspaces are the entire reason I like using gnome. When Gnome 40 releases I'll try to bring vertical workspaces back, but in the (likely) case I can't manage and no one else finds a way I'm switching to a different window manager and I wont be maintaining a fork of this anymore.

TLDR: This extensions is probably not surviving the Gnome 40 update and as such I don't think it's a good idea to upload this to the site.

**Comment by xakraz** [comment]
At least, can @passingthru67 merge this PR and release it on extensions.gnome.org ?

**Comment by horga83** [comment]
Many thanks for your work on this.  I do hope a way can be found for it to survive.  I live on Arch and fear my 5 years of Gnome is coming to a complete change.  I have tried KDE and didn't like it at all.  I really don't understand this change by the Gnome team as horizontal space is so valuable.   Lets hope we can have horizontal workspaces in a vertical app.  At any rate I wanted to jump in and pass along my thanks.

**Comment by RensAlthuis** [comment]
I come with good news, I've spend most of my free time this last week bringing vertical workspaces to gnome 40. It's coming along quite well. I'd like to bring back a lot of the functionality of workspaces-to-dock as wel, but that is a bigger project that may take a while.
For anyone interested in helping or just enjoying a preview of the vertically stacked workspaces: https://github.com/RensAlthuis/vertical-overview
(I'll make sure to add some gifs later)
